src/main/java/com/denizenscript/depenizen/bukkit/objects/residence/ResidenceTag.java

```diff
@@ -108,44 +114,73 @@ public String toString() {
 
     @Override
     public ObjectTag getObjectAttribute(Attribute attribute) {
-        if (attribute == null) {
-            return null;
-        }
+        return tagProcessor.getObjectAttribute(this, attribute);
+    }
+
+    public static void registerTags() {
 
         // <--[tag]
         // @attribute <ResidenceTag.name>
         // @returns ElementTag
         // @plugin Depenizen, Residence
         // @description
-        // Returns the name of the residence.
+        // Returns the name of this Residence.
         // -->
-        if (attribute.startsWith("name")) {
-            return new ElementTag(residence.getName()).getObjectAttribute(attribute.fulfill(1));
-        }
+        tagProcessor.registerTag(ElementTag.class, "name", (attribute, object) -> {
+            return new ElementTag(object.getResidence().getName());
+        });
 
         // <--[tag]
         // @attribute <ResidenceTag.owner>
         // @returns PlayerTag
         // @plugin Depenizen, Residence
         // @description
-        // Returns the owner of the residence.
+        // Returns a PlayerTag of owner in this Residence.
         // -->
-        else if (attribute.startsWith("owner")) {
-            return new PlayerTag(residence.getOwnerUUID()).getObjectAttribute(attribute.fulfill(1));
-        }
+        tagProcessor.registerTag(PlayerTag.class, "owner", (attribute, object) -> {
+            return new PlayerTag(object.getResidence().getOwnerUUID());
+        });
+
+        // <--[tag]
+        // @attribute <ResidenceTag.subzones>
+        // @returns ListTag
+        // @plugin Depenizen, Residence
+        // @description
+        // Returns a ListTag(ResidenceTag) of subzones in this Residence.
```

Comment:
Same here, the `returns` should specify what object type does the list include (`ListTag(ResidenceTag)`), and the `description` doesn't need to mention the exact type, I.e. something like  `Returns a list of subzones in this Residence.`